documentation/release_notes.markdown

```diff
@@ -10,6 +10,18 @@ canonical: "/puppetserver/latest/release_notes.html"
 [puppetserver.conf]: ./config_file_puppetserver.markdown
 [product.conf]: ./config_file_product.markdown
 
+## Puppet Server 6.12.0
+
+Released 3 June 2020
+
+### Resolved Issue
+
+- JRuby has been bumped to 9.2.11.1 again, with `invokedynamic.yield` set to false to resolve a stackoverflow error. [SERVER-2793](https://tickets.puppetlabs.com/browse/SERVER-2793)
+
+### Deprecation
```

Comment:
Wasn't sure which heading to use here, lmk if there's something else we've used in the past.

Reply:
"Deprecation" seems fine to me.